docs/integration_design/README.MD

```diff
@@ -0,0 +1,104 @@
+(We also require this documentation for our review and final inclusion in Demisto, but the former reason is the more important one).
+
+
+
+## Identify What Type of Integration I'm Building
+You can divide most integrations in Demisto into one of the following categories. We advise you to try to understand in which your integration fits in. Each link is to a short list of use-cases for such integration and examples of existing integrations and commands so that you can have some idea as to how or what to implement:
+## Usecases
+1. [Analytics and SIEM](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/Analytics_SIEM_usecase)
+2. [Authentication](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/Authentication_usecase)
+3. [Case Management](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/Case_Management_usecase)
+4. [Data Enrichment and Threat Intelligence](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/Data_Enrichment_Threat_Intelligence_usecase)
+5. [Email Gateway](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/Email_Gateway_usecase)
+6. [Endpoint](https://github.com/demisto/internal-content/blob/master/documentation/design/usecases/Endpoint_usecase/README.MD)
+7. [Forensics and Malware](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/forensics_malware_usecase)
+8. [Network Security (Firewall)](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/network_security_IDS_IPS_usecase)
+9. [Network Security (IDS/IPS)](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/network_security_firewall_usecase)
+10. [Vulnerability Management](https://github.com/demisto/content/tree/master/docs/integration_design/usecases/vulnerability_management_usecase)
+
+
+
+## Understanding and Documenting the Use Cases
+If your'e on your way to writing an integration for Demisto you probably are already familiar with the product you want to integrate, but it might help to go into the product again to see what actions you want to implemnt, how you think they'll work once in Demisto.
+Write down the use cases for your integration, how you would use it and how you think outher people will use it.
+```Markdown
+# Use Cases:
+- Create, get, edit, close a ticket/issue, add + view comments.
+- Assign a ticket/issue to a specified user.
+- List all tickets, filter by name, date, assignee
+```
+## Parameters
+In this section write what are the parameters you need in order to create an instance of the product and connect to it. These parameters will be accessible and used in all the integration. 
+We require 2 fields as part of the parameters:
```

Comment:
mention common parameters that we see in integrations, like `url`, `apikey` / `apitoken`/ `credentials (username, password)`, if the integrations fetches incidents then `query` which will query events/incidents/tickets